client/src/util/shallowEqual.js

```diff
@@ -0,0 +1,78 @@
+/*
```

Comment:
I think `react-redux` has `shallowEqual`, so maybe we can use that one?

`import { shallowEqual } from 'react-redux'`

source: https://react-redux.js.org/next/api/hooks#equality-comparisons-and-updates

Reply:
good suggestion - will make this change.

Reply:
note: this will make lots of our util modules dependent on react-redux, even though they have nothing to do with redux.  But I think that is a reasonable trade.